src/content/en/2021/caching.md

```diff
@@ -17,4 +19,328 @@ featured_stat_3: TODO
+Browser: It will usually either learn how to cache your resources internally or respect the HTTP Headers you defined to optimize the experience. On top of it, you will have access to several more manual caching strategies either by storing simple strings in Cookies, complex API responses in IndexdDB, or entire JS or HTML resources in the CacheStorage with the Service Worker.
+
+The three main pillars of how we cache web resources are:
+- Cache as much as you can
+- Cache for as long as you can
+- Cache as close to end users as you can
+
+## CDN Cache adoption
+
+A Content Delivery Network (CDN) is a group of servers spread out over several locations that store copies of data. This allows servers to fulfill requests based on the server closest to the end-user. (Remember we want to cache as close to end-users as possible.) In 2021 across the web, the most popular CDN for Desktop was Cloudflare with 14% of pages, followed by Google at 6% adoption. While Cloudflare is used twice as much as Google, a large variety of solutions remains available, including Fastly, Amazon CloudFront, Akamai, and many others.
+
+{{ figure_markup(
+  image="top-cdns.png",
+  caption="Adoption of the top CDNs.",
+  description="TODO",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=2035524639&format=interactive",
+  height=501,
+  sheets_gid="58739923",
+  sql_file="top_cdns.sql"
+) }}
+
+## Service Worker adoption
+
+A service worker is a script that the browser allows to run in the background independently from a web page. It supports features that don't need direct user or web page interaction and offer rich offline experiences.
+
+The adoption of Service Workers has continued to steadily increase. While 1% of Mobile pages registered a service worker in 2020, 9% of Mobile pages ranked in the top 1,000 registered one in 2021.
+
+{{ figure_markup(
+  image="sw-adoption.png",
+  caption="Service Worker adoption.",
+  description="TODO",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=688062455&format=interactive",
+  sheets_gid="802609299",
+  sql_file="service_worker_rank.sql"
+) }}
+
+The primary way to cache resources within a Service Worker, it's by using the CacheStorage API. This allows a developer to create a custom cache strategy for any requests passing through the worker; some famous ones are Stale-While-Revalidate, Cache Falling Back to Network, Network Falling Back to Cache, Cache Only. In recent years become even easier to adopt those strategies by increasing the popularity of [Workbox](https://developers.google.com/web/tools/workbox/modules/workbox-strategies), which decides what cache you want to plug and play.
```

Comment:
Edits:

The primary way to cache resources within a Service Worker is by using the CacheStorage API. This allows a developer to create a custom cache strategy for any requests passing through the worker; some well-known ones are Stale-While-Revalidate, Cache Falling Back to Network, Network Falling Back to Cache, and Cache Only. In recent years it has become even easier to adopt those strategies by increasing the popularity of [Workbox](https://developers.google.com/web/tools/workbox/modules/workbox-strategies), which decides what cache you want to plug and play with.